src/wcs1_0/src/main/java/org/geoserver/wcs/kvp/Wcs10GetCoverageRequestReader.java

```diff
@@ -389,13 +394,35 @@ private RangeSubsetType parseRangeSubset(Map kvp, String coverageName) {
             } else checkTypeAxisRange(rangeSubset, axis, "ELEVATION");
         }
 
+        // check for custom dimensions
+        CoverageInfo coverage = this.catalog.getCoverageByName(coverageName);
+        if (coverage != null) {
+            for (Entry<String, ?> entry : coverage.getMetadata().entrySet()) {
+                String name = entry.getKey();
+                if (name.startsWith(ResourceInfo.CUSTOM_DIMENSION_PREFIX)
+                        && entry.getValue() instanceof DimensionInfo
+                        && ((DimensionInfo) entry.getValue()).isEnabled()) {
+                    name = name.substring(ResourceInfo.CUSTOM_DIMENSION_PREFIX.length());
+                    Object value = kvp.get(name);
+                    // ignore anything that got converted by a KVP parser so that they
```

Comment:
Best to use the rawKVP then, it contains the unparsed values.

Reply:
It is intentional that kvp is used here instead of rawKVP.  This assumes that anything that got parsed from a String is a keyword and should be ignored as a custom dimension so that it isn't being handled both as a keyword and a custom dimension.

Reply:
Hmm... wondering about the WMS convention of prefixing with DIM_ to avoid conflicts (since we are in "vendor behavior" territory anyways). For WCS 2.0 there is no issue as the name is not used a key, did you check what WCS 1.1 is doing, if anything at all?

Reply:
As far as I can tell, WCS 1.1 only handles the time dimension.  It does not have elevation or custom dimension support implemented even for XML requests.

Unlike WMS, the DIM_ prefix isn't in the WCS 1.0 specification but I could definitely add it if it would help.

Reply:
Are custom dimensions part of the WCS 1.0 specification at all?

Reply:
Yes, it is part of the specification although GeoServer's WCS 1.0 DescribeCoverage doesn't actually show the custom dimensions.
> If the range set of the selected coverage consists of compound values, GetCoverage requests may include constraints defined on the parameter(s) of the compound range set. Such constraints are expressed as “PARAMETER=[value]”, where  
a) The variable string PARAMETER matches the name of an AxisDescription element defined on that range set, and  
b) [value] is one of the acceptable values defined in the corresponding AxisDescription element.

Reply:
Oh ok, then let's not change the current code.